### PR TITLE
Make selected item get focus when dropdown is opened

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

As it stands, dropdowns currently do not focus the item that is selected, so if you select something on a dropdown, and then close it, and re-open it, then the new item is not auto-focused. This PR changes that so that selected value is focused by default when the dropdown is re-opened.

## Tests

- Added tests for making sure that the right element is focused when a dropdown opens, and that the dropdown itself is focused when the dropdown is closed.

## Breaking Change

- [X] Yes, this is a breaking change, because it changes the focus behavior of dropdowns. Only dropdowns that expect to control focus themselves are affected.

### CODE REVIEW

File: packages/flutter/lib/src/material/dropdown.dart [hunk 1]
```diff
@@ -520,7 +530,7 @@ class _DropdownRoutePage<T> extends StatelessWidget {
 // each menu item. These sizes are used to compute the offset of the selected
 // item so that _DropdownRoutePage can align the vertical center of the
 // selected item lines up with the vertical center of the dropdown button,
-// as closely as posible.
+// as closely as possible.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown.dart]: NICE

File: packages/flutter/test/material/dropdown_test.dart [hunk 2]
```diff
@@ -1993,4 +1996,129 @@ void main() {
...
+    await tester.sendKeyEvent(LogicalKeyboardKey.enter);
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1)); // finish the menu open animation
+    expect(value, equals('two'));
+    final Element element =tester.element(find.byKey(const ValueKey<String>('two')).last);
+    final FocusNode node = Focus.of(element);
+    expect(node.hasFocus, isTrue);
+  });
+
+  testWidgets('Selected element is correctly focused with a large dropdown', (WidgetTester tester) async {
```
Comment (HansMuller) [inline packages/flutter/test/material/dropdown_test.dart]: Note that it matters but: the point of this is to verify that we DTRT with a scrollable dropdown. "Large" (and "huge") sound like a consequence of a large value for textScaleFactor.

File: packages/flutter/test/material/dropdown_test.dart [hunk 3]
```diff
@@ -1993,4 +1996,129 @@ void main() {
...
+    await tester.sendKeyEvent(LogicalKeyboardKey.enter); // Select '42' and close the dropdown.
+    await tester.pumpAndSettle(const Duration(seconds: 1)); // Finish the menu close animation
+    expect(value, equals(42));
+
+    // Now make sure that "42" is focused when we re-open the dropdown.
+    await tester.sendKeyEvent(LogicalKeyboardKey.enter);
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1)); // finish the menu open animation
+    expect(value, equals(42));
+    final Element element =tester.element(find.byKey(const ValueKey<int>(42)).last);
```
Comment (HansMuller) [inline packages/flutter/test/material/dropdown_test.dart]: Missing a space `= tester...`

File: packages/flutter/test/material/dropdown_test.dart [hunk 4]
```diff
@@ -1993,4 +1996,129 @@ void main() {
...
+
+    await tester.sendKeyEvent(LogicalKeyboardKey.enter);
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1)); // finish the menu open animation
+    expect(value, equals(1));
+    expect(Focus.of(tester.element(find.byKey(const ValueKey<int>(1)).last)).hasPrimaryFocus, isTrue);
+
+    for (int i = 0; i < 41; ++i) {
+      await tester.sendKeyEvent(LogicalKeyboardKey.tab); // Move to the next one.
+      await tester.pumpAndSettle(const Duration(milliseconds: 200)); // Wait for it to animate the menu.
```
Comment (HansMuller) [inline packages/flutter/test/material/dropdown_test.dart]: Is the duration parameter really needed here? It doesn't jibe with the 100ms animateTo duration and (I think maybe) the test would be equally valid without it.

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/test/material/dropdown_test.dart (hunk 2 above)
Comment (gspencergoog) [reply packages/flutter/test/material/dropdown_test.dart]: OK, I'll change it to "with a lot of items".

File: packages/flutter/test/material/dropdown_test.dart (hunk 4 above)
Comment (gspencergoog) [reply packages/flutter/test/material/dropdown_test.dart]: Removed the duration. You're right, it's irrelevant.

File: packages/flutter/test/material/dropdown_test.dart (hunk 3 above)
Comment (gspencergoog) [reply packages/flutter/test/material/dropdown_test.dart]: Whoops. Done.

**Pull request merged**